source/shader/StarField.cpp

```diff
@@ -125,11 +126,38 @@ void StarField::SetHaze(const Sprite *sprite, bool allowAnimation)
 
 
 
-void StarField::Draw(const Point &pos, const Point &vel, double zoom, const System *system) const
+void StarField::Step(Point vel, double zoom)
 {
-	double density = system ? system->StarfieldDensity() : 1.;
+	if(Preferences::Has("Fixed starfield zoom"))
+	{
+		baseZoom = constants->GetValue("fixed zoom");
+		vel /= constants->GetValue("velocity reducer");
```

Comment:
Maybe move this to Init to avoid doing these three lookups every frame.

Reply:
We'd have to make sure this interface is loaded prior to the init call.

Reply:
You could add a "FinishLoading" method to the class that retrieves and stores these values after all the sources of game data have been parsed.